rotate-preservation-system-password/src/main/scala/uk/gov/nationalarchives/rotatepreservationsystempassword/Lambda.scala

```diff
@@ -0,0 +1,111 @@
+package uk.gov.nationalarchives.rotatepreservationsystempassword
+
+import cats.effect.IO
+import io.circe.{Decoder, HCursor}
+import pureconfig.ConfigReader
+import pureconfig.generic.derivation.default.*
+import software.amazon.awssdk.services.secretsmanager.model.DescribeSecretResponse
+import uk.gov.nationalarchives.DASecretsManagerClient
+import uk.gov.nationalarchives.DASecretsManagerClient.Stage
+import uk.gov.nationalarchives.DASecretsManagerClient.Stage.*
+import uk.gov.nationalarchives.dp.client.UserClient
+import uk.gov.nationalarchives.dp.client.UserClient.ResetPasswordRequest
+import uk.gov.nationalarchives.dp.client.fs2.Fs2Client.userClient
+import uk.gov.nationalarchives.rotatepreservationsystempassword.Lambda.{*, given}
+import uk.gov.nationalarchives.rotatepreservationsystempassword.Lambda.RotationStep.*
+import uk.gov.nationalarchives.utils.LambdaRunner
+
+import scala.jdk.CollectionConverters.*
+
+class Lambda extends LambdaRunner[RotationEvent, Unit, Config, Dependencies] {
+
+  override def dependencies(config: Config): IO[Dependencies] = userClient(config.apiUrl, config.secretName).map { userClient =>
+    Dependencies(userClient, secretId => DASecretsManagerClient[IO](secretId))
+  }
+
+  override def handler: (RotationEvent, Config, Dependencies) => IO[Unit] = (event, config, dependencies) =>
+    for {
+      client <- IO.pure(dependencies.secretsManagerClient(event.secretId))
+      token = event.clientRequestToken
+      secretId = event.secretId
+      describeSecretResponse <- client.describeSecret()
+      versions <- versions(describeSecretResponse)
+      _ <- IO.raiseWhen(!describeSecretResponse.rotationEnabled())(new Exception(s"Secret $secretId is not enabled for rotation"))
+      _ <- IO.raiseWhen(!versions.contains(event.clientRequestToken))(new Exception(s"Secret version $token has no stage for rotation of secret $secretId."))
```

Comment:
This message is a little unclear, maybe `"Secret version $token has no stage set for rotation of secret $secretId."` or `"No stage has been set for secret version $token for rotation of secret $secretId."`?

Reply:
I've gone with "Secret version $token has no stage set for rotation of secret $secretId." :+1: